.github/workflows/main.yml

```diff
@@ -1,40 +1,72 @@
 ---
 name: CI
 
+env:
+  IMAGE_NAME: research-template
+  PUBLIC_IMAGE_NAME: ghcr.io/opensafely/research-template
+  REGISTRY: ghcr.io
+
 on:
+  workflow_dispatch:
+  pull_request:
   push:
+    branches: [main]
 
 jobs:
-  check:
+  build:
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
+      - uses: hadolint/hadolint-action@54c9adbab1582c2ef04b2016b760714a4bfde3cf  # v3.1.0
+        # Lint the dockerfile before building
+        with:
+          failure-threshold: error
```

Comment:
question: Do we still need this extra configuration to disable checks on warnings, after ignoring the warnings?

Reply:
It's not disabling the checks, it just stops it failing on warnings and below. The default is to fail on info level checks. 

It's interesting that we don't have it configured like this elsewhere because it's in the repo-template! I don't know why the threshold was set to error there and not elsewhere though.